DatabaseBuddy.Core/DatabaseExtender/DBConnection.cs

```diff
@@ -13,30 +14,37 @@ public class DBConnection
         private string m_DatabaseName;
         private string m_Username;
         private string m_Password;
+        private bool m_IntegratedSecurity;
         public event EventHandler ConnectionFailed;
 
         public DBConnection(string ServerName, string DatabaseName,
-            string Username, string Password)
+            string Username, string Password, bool IntegratedSecurity = false)
```

Comment:
Better use overloaded constructor, because `Username` and `Password` is only required if `IntegratedSecurity`is false.